src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1624,32 +1644,39 @@ uint PhaseChaitin::Select( ) {
     }
 
     // Check if a color is available and if so pick the color
-    OptoReg::Name reg = choose_color( *lrg, chunk );
+    OptoReg::Name reg = choose_color(*lrg);
 
     //---------------
     // If we fail to color and the AllStack flag is set, trigger
     // a chunk-rollover event
-    if(!OptoReg::is_valid(OptoReg::add(reg,-chunk)) && is_allstack) {
+    if (!OptoReg::is_valid(reg) && is_allstack) {
       // Bump register mask up to next stack chunk
-      chunk += RegMask::CHUNK_SIZE;
-      lrg->Set_All();
+      bool success = lrg->rollover();
+      if (!success) {
```

Comment:
Was this scenario (running out of stack slots representable in `OptoRegPairs`) possible before, or was it prevented by some check removed in the changeset? Did you come across it in some compilation or is it more of a "theoretical" guard?

Reply:
Yes, it is a theoretical guard (also see the discussions earlier in this PR) and could also happen before this changeset if we roll over too much in `Select`. I experimented a bit with this earlier on and was not able to construct an example where we end up in this situation.